### PR TITLE
Dlukeomalley/integrations

### PR DESCRIPTION
**Pull request by dlukeomalley**

### CODE REVIEW

File: docs/integrations.md [hunk 1]
```diff
@@ -11,11 +11,119 @@ Semgrep integrations exist from the beginning to the end of the development flow
...
+<p>
+
+```yaml
+hello: {}
+```
 
+</p>
+</details>
+</br>
 Is your CI provider missing? Let us know by [filing an issue here](https://github.com/returntocorp/semgrep/issues/new?assignees=&labels=&template=feature_request.md&title=).
```
Comment (ievans) [inline docs/integrations.md:127]: @dlukeomalley this link doesn't render

**Comment by ievans** [comment]
@dlukeomalley Overall seems good, but perhaps the docs could mention earlier (ie before the YAML comments) that by setting up the semgrep.dev account, you will get the SEMGREP_DEPLOYMENT_ID and APP TOKEN to populate a secrets store. Or we should have a separate PR to semgrep action in the case that those variables are not populated -- we point the user at semgrep.dev/manage

Also, it is a bit unclear that you ought to *not* put the app token in directly to the YAML, but rather let them be populated by env vars. I feel like someone new to the actions workflow could easily make that mistake.

File: docs/integrations.md [hunk 2]
```diff
@@ -11,11 +11,119 @@ Semgrep integrations exist from the beginning to the end of the development flow
...
+    executor:
+      docker:
+        image: returntocorp/semgrep-agent:v1
+        entrypoint: semgrep-agent
+        command: [
+          "--publish-deployment",
+          "$SEMGREP_DEPLOYMENT_ID",
+          "--publish-token", 
+          "$SEMGREP_APP_TOKEN"
+        ]
```
Comment (dlukeomalley) [inline docs/integrations.md:32]: This need verification

File: docs/integrations.md (hunk 1 above)
Comment (dlukeomalley) [reply docs/integrations.md:127]: When you say "doesn't render", does it not open for you or the ticket it opens isn't clear?

Comment (ievans) [reply docs/integrations.md:112]: On the github markdown preview I see this: Screen Shot 2020-10-26 at 9 33 54 PM

Comment (dlukeomalley) [reply docs/integrations.md:112]: Ah, roger. This is not valid GitHub markdown (requires mkdocs compilation) 👍

File: docs/index.md [hunk 3]
```diff
@@ -29,12 +29,12 @@ Semgrep rules generally look like the code you’re searching for. For example,
 
 The [Semgrep registry](https://semgrep.dev/explore) has 900+ rules written by the Semgrep team as well as community contributors covering security, correctness, and performance bugs. No need to DIY unless you want to.
 
-Semgrep is an evolution of [pfff](https://github.com/returntocorp/pfff/), which began at [Facebook](https://github.com/facebookarchive/pfff) in 2009, which itself was an evolution of the Linux refactoring tool [Concinelle](https://en.wikipedia.org/wiki/Coccinelle_(software)). r2c revitalized the project after its original author, Yoann Padioleau, joined the company.
+Semgrep is an evolution of [pfff](https://github.com/returntocorp/pfff/), which began at [Facebook](https://github.com/facebookarchive/pfff) in 2009, which itself was an evolution of the Linux refactoring tool [Concinelle](https://en.wikipedia.org/wiki/Coccinelle_(software)). r2c revitalized the project after its original author, [Yoann Padioleau](https://github.com/aryx), joined the company.
```
Comment (underyx) [inline docs/index.md:32]:
```suggestion
Semgrep is an evolution of [pfff](https://github.com/returntocorp/pfff/), which began at [Facebook](https://github.com/facebookarchive/pfff) in 2009, which itself was an evolution of the Linux refactoring tool [Coccinelle](https://en.wikipedia.org/wiki/Coccinelle_(software)). r2c revitalized the project after its original author, [Yoann Padioleau](https://github.com/aryx), joined the company.
```

Typo in Coccinelle

File: docs/integrations.md [hunk 4]
```diff
@@ -1,21 +1,117 @@
...
-Semgrep integrations exist from the beginning to the end of the development flow, from code conception in the IDE to code review and beyond in pull requests, Slack, over email, and more. Workflows vary widely between teams and their projects.
+Semgrep integrates into the development flow end-to-end, from code conception in the IDE to code review and beyond in pull requests, Slack, over email, and more. Everyone's workflow is a little different and Semgrep is meant to adapt to yours.
 
 [TOC]
 
 # Continuous integration (CI)
 
-!!! info
-    [Semgrep CI](https://github.com/returntocorp/semgrep-action) is a Git and diff-aware Semgrep wrapper that surfaces only the results introduced by a pull request. With this approach teams can enable new checks without being penalized for pre-existing issues 👏
+The following instructions use [Semgrep CI](https://github.com/returntocorp/semgrep-action) and require a free [Semgrep Community](https://semgrep.dev/manage) or paid Semgrep Team account. `SEMGREP_DEPLOYENT_ID` and `SEMGREP_APP_TOKEN` information is available at [Manage > Projects](https://semgrep.dev/manage/projects) after login.
```
Comment (underyx) [inline docs/integrations.md:9]:
```suggestion
The following instructions use [Semgrep CI](https://github.com/returntocorp/semgrep-action) and require a free [Semgrep Community](https://semgrep.dev/manage) or paid Semgrep Team account. `SEMGREP_DEPLOYMENT_ID` and `SEMGREP_APP_TOKEN` information is available at [Manage > Projects](https://semgrep.dev/manage/projects) after login.
```

File: docs/integrations.md [hunk 5]
```diff
@@ -1,21 +1,117 @@
...
+        docker:
+            - image: returntocorp/semgrep-agent:v1
+        steps:
+            - checkout
+            - run: python -m semgrep_agent --publish-deployment $SEMGREP_DEPLOYMENT_ID --publish-token $SEMGREP_APP_TOKEN
+```
+
+</p>
+</details>
+<details><summary>GitHub Action</summary>
```
Comment (underyx) [inline docs/integrations.md:50]:
```suggestion
<details><summary>GitHub Actions</summary>
```

File: docs/integrations.md [hunk 6]
```diff
@@ -1,21 +1,117 @@
...
+```yaml
+name: Semgrep
+
+on: 
+    # Run on all pull requests. Returns the results introduced by the PR.
+    pull_request: {}
+
+    # Run on merges. Returns all results.
+    #push:
+    #    branches: ["master"]
```
Comment (underyx) [inline docs/integrations.md:62]:
```suggestion
    #    branches: ["master", "main"]
```

Main is now the default branch name.

File: docs/integrations.md [hunk 7]
```diff
@@ -1,21 +1,117 @@
...
+      #  uses: github/codeql-action/upload-sarif@v1
+      #  with:
+      #    sarif_file: semgrep.sarif
+      #  if: always()
+```
 
-Instructions are available at [semgrep.dev/manage](https://semgrep.dev/manage) as part of the “setup a new project” workflow. Those instructions offer provider-specific configurations for [Semgrep CI](https://github.com/returntocorp/semgrep-action) for the following products:
+</p>
+</details>
+<details><summary>GitLab Jobs</summary>
```
Comment (underyx) [inline docs/integrations.md:90]:
```suggestion
<details><summary>GitLab CI</summary>
```

**Review by underyx — APPROVED** [review]

**Pull request merged**